packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -191,11 +197,11 @@ export default function VisualEditor( { styles } ) {
 
 	const titleRef = useRef();
 	useEffect( () => {
-		if ( isWelcomeGuideVisible ) {
+		if ( isWelcomeGuideVisible || ! isCleanNewPost ) {
 			return;
 		}
 		titleRef?.current?.focus();
-	}, [ isWelcomeGuideVisible ] );
+	}, [ isWelcomeGuideVisible, isCleanNewPost ] );
```

Comment:
A side effect of adding the value of `isCleanNewPost` as a dependency here is that the effect will run if `isCleanNewPost` changes even if `isWelcomeGuideVisible` didn't change. Meaning if the post was not clear but becomes clean (could happen programmatically for instance) while the welcome guide is open, the title will be focused and the welcome guide closes (focus moved outside).

The probably of the above happening is small of course but I think there's a better way to implement this logic. (Calling the selector inside the effect).


So you'd select the selector by doing :

```
const { isCleanNewPost } = useSelect( editorStore );
```

and then in the effect, you'd replace ` ! isCleanNewPost` with ` ! isCleanNewPost()` instead.

Note that this technique of calling selectors outside of `useSelect` should only be used if the returned value is not used in the rendering of the React tree but in an effect or event handler or any random callback.

Reply:
@youknowriad I get this error.

> React Hook "useSelect" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function.

What should I do?
1. Ignore it with a comment.
2. Add a comment explaining this effect will only fire if post is clean as defined in editor. Then not use isCleanNewPost in edit-post and leave that functionality in editor.
3. Other ideas?

Reply:
Actually, I guess it will not work. I generated a local build and this was logged in console.

```
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
```

Reply:
The useSelect call shouldn't be inside the callback but it should just replace the existing one line 116

Reply:
Ah, okay. Fixed. Just the 1 call to `useSelect( store_name );` retrieves once, yes? That way, once it is retrieved, it will not trigger the effect again?